backend/src/test/java/fk/prof/backend/AssociationApiTest.java

```diff
@@ -0,0 +1,239 @@
+    );
+
+    boolean released = latch.await(10, TimeUnit.SECONDS);
+    if (!released) {
+      context.fail("Latch timed out but leader election task was not run");
+    }
+    //This sleep should be enough for leader discovery store to get updated with the new leader
+    Thread.sleep(1500);
+
+    //Leader has been elected, it will be same as backend, since backend verticles were not undeployed
+    Recorder.ProcessGroup processGroup = Recorder.ProcessGroup.newBuilder().setAppId("a").setCluster("c").setProcName("p1").build();
+    makeRequestGetAssociation(processGroup).setHandler(ar -> {
+      if(ar.succeeded()) {
+        context.assertEquals(400, ar.result().getStatusCode());
+        async.complete();
+      } else {
+        context.fail(ar.cause());
+      }
+    });
+  }
+
+  /**
+   * Tests following scenario:
+   * => association is requested from backend, proxied to leader, which returns 500 because no backends are known to leader
+   * => one backend reports its load to leader
+   * => association is requested again, proxied to leader, returns the backend which reported its load earlier
+   * @param context
+   * @throws InterruptedException
+   */
+  @Test
+  public void getAssociationProxiedToLeader(TestContext context) throws InterruptedException {
+    final Async async = context.async();
+    CountDownLatch latch = new CountDownLatch(1);
+    Runnable leaderElectedTask = VertxManager.getDefaultLeaderElectedTask(
+        vertx, true, null, true,
+        ConfigManager.getLeaderHttpServerConfig(config),
+        new DeploymentOptions(ConfigManager.getLeaderHttpDeploymentConfig(config)),
+        backendAssociationStore);
+
+    VertxManager.deployLeaderElectionWorkerVerticles(
+        vertx,
+        new DeploymentOptions(ConfigManager.getLeaderElectionDeploymentConfig(config)),
+        curatorClient,
+        leaderElectedTask,
+        leaderDiscoveryStore
+    );
+
+    //This sleep should be enough for leader discovery store to get updated with the new leader and leader elected task to be executed
+    Thread.sleep(5000);
+    when(leaderDiscoveryStore.isLeader()).thenReturn(false);
+
+    //Leader has been elected, it will be same as backend, since backend verticles were not undeployed
+    Recorder.ProcessGroup processGroup = Recorder.ProcessGroup.newBuilder().setAppId("a").setCluster("c").setProcName("p1").build();
+    makeRequestGetAssociation(processGroup).setHandler(ar -> {
+      if(ar.succeeded()) {
+        context.assertEquals(500, ar.result().getStatusCode());
+        makeRequestReportLoad(BackendDTO.LoadReportRequest.newBuilder().setIp("1").setLoad(0.5f).build())
+            .setHandler(ar1 -> {
+              context.assertTrue(ar1.succeeded());
+              makeRequestGetAssociation(processGroup).setHandler(ar2 -> {
+                context.assertTrue(ar2.succeeded());
+                System.out.println(ar2.result());
+                context.assertEquals(200, ar2.result().getStatusCode());
```

Comment:
Something should verify that this callback was actually called, and others weren't.

Here if this lambda isn't called, all asserts will actually be bypassed.